olp-cpp-sdk-core/tests/cache/DefaultCacheImplTest.cpp

```diff
@@ -322,6 +323,48 @@ TEST(DefaultCacheImplTest, LruCacheRemove) {
   }
 }
 
+TEST(DefaultCacheImplTest, ProtectedCacheExpired) {
+  SCOPED_TRACE("Expiry protected cache");
+  const std::string key1{"somekey1"};
+  const std::string key2{"somekey2"};
+  const std::string data_string{"this is key's data"};
+  constexpr auto expiry = 1;
+  std::vector<unsigned char> binary_data = {1, 2, 3};
+  const auto data_ptr =
+      std::make_shared<std::vector<unsigned char>>(binary_data);
+
+  std::string cache_path = olp::utils::Dir::TempDirectory() + "/unittest";
```

Comment:
I wonder if this is safe? I don't see this path being removed anywhere. And the next test just picks up the same dirty cache and uses the same keys also. Please delete the folder every time or make a helper class that automatically deletes when out of scope.